src/ICat.hs

```diff
@@ -1,332 +1,59 @@
-  , ("151", "afd7af")
-  , ("152", "afd7d7")
-  , ("153", "afd7ff")
-  , ("154", "afff00")
-  , ("155", "afff5f")
-  , ("156", "afff87")
-  , ("157", "afffaf")
-  , ("158", "afffd7")
-  , ("159", "afffff")
-  , ("160", "d70000")
-  , ("161", "d7005f")
-  , ("162", "d70087")
-  , ("163", "d700af")
-  , ("164", "d700d7")
-  , ("165", "d700ff")
-  , ("166", "d75f00")
-  , ("167", "d75f5f")
-  , ("168", "d75f87")
-  , ("169", "d75faf")
-  , ("170", "d75fd7")
-  , ("171", "d75fff")
-  , ("172", "d78700")
-  , ("173", "d7875f")
-  , ("174", "d78787")
-  , ("175", "d787af")
-  , ("176", "d787d7")
-  , ("177", "d787ff")
-  , ("178", "d7af00")
-  , ("179", "d7af5f")
-  , ("180", "d7af87")
-  , ("181", "d7afaf")
-  , ("182", "d7afd7")
-  , ("183", "d7afff")
-  , ("184", "d7d700")
-  , ("185", "d7d75f")
-  , ("186", "d7d787")
-  , ("187", "d7d7af")
-  , ("188", "d7d7d7")
-  , ("189", "d7d7ff")
-  , ("190", "d7ff00")
-  , ("191", "d7ff5f")
-  , ("192", "d7ff87")
-  , ("193", "d7ffaf")
-  , ("194", "d7ffd7")
-  , ("195", "d7ffff")
-  , ("196", "ff0000")
-  , ("197", "ff005f")
-  , ("198", "ff0087")
-  , ("199", "ff00af")
-  , ("200", "ff00d7")
-  , ("201", "ff00ff")
-  , ("202", "ff5f00")
-  , ("203", "ff5f5f")
-  , ("204", "ff5f87")
-  , ("205", "ff5faf")
-  , ("206", "ff5fd7")
-  , ("207", "ff5fff")
-  , ("208", "ff8700")
-  , ("209", "ff875f")
-  , ("210", "ff8787")
-  , ("211", "ff87af")
-  , ("212", "ff87d7")
-  , ("213", "ff87ff")
-  , ("214", "ffaf00")
-  , ("215", "ffaf5f")
-  , ("216", "ffaf87")
-  , ("217", "ffafaf")
-  , ("218", "ffafd7")
-  , ("219", "ffafff")
-  , ("220", "ffd700")
-  , ("221", "ffd75f")
-  , ("222", "ffd787")
-  , ("223", "ffd7af")
-  , ("224", "ffd7d7")
-  , ("225", "ffd7ff")
-  , ("226", "ffff00")
-  , ("227", "ffff5f")
-  , ("228", "ffff87")
-  , ("229", "ffffaf")
-  , ("230", "ffffd7")
-  , ("231", "ffffff")
-  , ("232", "080808")
-  , ("233", "121212")
-  , ("234", "1c1c1c")
-  , ("235", "262626")
-  , ("236", "303030")
-  , ("237", "3a3a3a")
-  , ("238", "444444")
-  , ("239", "4e4e4e")
-  , ("240", "585858")
-  , ("241", "626262")
-  , ("242", "6c6c6c")
-  , ("243", "767676")
-  , ("244", "808080")
-  , ("245", "8a8a8a")
-  , ("246", "949494")
-  , ("247", "9e9e9e")
-  , ("248", "a8a8a8")
-  , ("249", "b2b2b2")
-  , ("250", "bcbcbc")
-  , ("251", "c6c6c6")
-  , ("252", "d0d0d0")
-  , ("253", "dadada")
-  , ("254", "e4e4e4")
-  , ("255", "eeeeee")
-  ]
-
-
-intToHex :: (Integral a, Show a) => a -> [Char]
-intToHex i =
-  let
-    h = (Numeric.showHex i "")
-  in
-    if i <= 16 then
-     '0' : h
-    else
-      h
-
-
+-- import qualified System.Directory             as Directory
+import qualified System.Environment           as Environment
+
+intToHex :: (Integral a, Show a) => a -> String
+intToHex i
+  | i <= 16   = '0': h
+  | otherwise = h
+  where h = Numeric.showHex i ""
+
+-- you write code like you have to do work upfront
+-- before creating a reference to a computation.
+-- no need for that.
```

Comment:
This is sort of the frustrating thing about having the two choices between `let` and `where`. It boils down to style I guess. I do like what you've done with `approx'` though. It looks much better with the guards and the where style.